libraries/chain/include/eosio/chain/resource_limits.hpp

```diff
@@ -48,7 +46,12 @@ namespace resource_limits {
       int64_t used = 0; ///< quantity used in current window
       int64_t available = 0; ///< quantity available in current window (based upon fractional reserve)
       int64_t max = 0; ///< max per window under current congestion
+      int64_t staked_virtual_balance = 0;
```

Comment:
propose to add an additional bandwidth limits for users with insufficient ram-staked tokens by calculating the virtual value of user's resources. for these users it is negative for ram, and >= 0 for bandwidths; total value can't be negative for a successful transaction.